datasette/static/sort_utils.js

```diff
@@ -0,0 +1,142 @@
+function sort() {
+  /*
+    Generate url with sorting parameters according to the 
+    sorting queue
+  */
+  let sq = window.sorting_queue;
+  if (!sq.length) {
+    if (!document.getElementById(`alert_cols`))
+      document
+        .getElementById("sort_utils")
+        .insertAdjacentHTML(
+          "afterend",
+          `<p id='alert_cols'>You need to select some columns to sort!</p>`
+        );
+    return;
+  }
+  var sort_url = new URLSearchParams(window.search);
+  for (let option of sq) {
+    sort_url.append(
+      `_sort${option.direction === "dsc" ? "_desc" : ""}`,
+      option.name
+    );
+  }
+  //Taken from table.js - line 37
+  window.location.href = sort_url ? "?" + sort_url : location.pathname;
+}
+
+function toggleSortMenu() {
+  /*
+    Function used to toggle the visibility of the sorting menu
+  */
+  let menu = document.getElementById("sort_menu");
+  let btn = document.getElementById("toggle_sort_menu");
+  if (menu.style.display == "none") {
+    menu.style.display = "inline-block";
+    menu.classList.add("anim-scale-in");
+    //Taken from table.js - lines 79-85
+    document.addEventListener("click", function (ev) {
+      var target = ev.target;
+      while (target && target != menu && target != btn) {
+        target = target.parentNode;
+      }
+      if (!target) {
+        menu.style.display = "none";
+      }
+    });
+  } else {
+    menu.style.display = "none";
+  }
```

Comment:
nit: do an early return to avoid one level of indentation, making this more readable.
```suggestion
  if (menu.style.display != "none") { 
    menu.style.display = "none";
    return;
  }
  menu.style.display = "inline-block";
  menu.classList.add("anim-scale-in");
  //Taken from table.js - lines 79-85
  document.addEventListener("click", function (ev) {
    var target = ev.target;
    while (target && target != menu && target != btn) {
      target = target.parentNode;
    }
    if (!target) {
      menu.style.display = "none";
    }
  });
```